probe-rs/src/probe.rs

```diff
@@ -420,6 +421,20 @@ impl Probe {
         }
     }
 
+    /// Configure the scan chain to use for the attached target.
+    ///
+    /// See [`DebugProbe::set_scan_chain`] for more information and usage
+    pub fn set_scan_chain(
+        &mut self,
+        scan_chain: Vec<ScanChainElement>,
+    ) -> Result<(), DebugProbeError> {
+        if !self.attached {
```

Comment:
In the description of this method in the Trait, you write that one has to be attched to call this method, yet, here you do the opposite check. Am I misunderstanding something?

Reply:
No. you're absolutely right. That's embarrassing it looks like when cleaning up I did two things:

1. Changed the wrong function `select_protocol`. That shouldn't be changed in this PR at all.
1. Didn't change an existing bug in `set_scan_chain` which had the wrong logic as you pointed out.